pkg/core/native/native_neo.go

```diff
@@ -828,6 +835,34 @@ func (n *NEO) registerCandidate(ic *interop.Context, args []stackitem.Item) stac
 	return stackitem.NewBool(err == nil)
 }
 
+func (n *NEO) onNEP17Payment(ic *interop.Context, args []stackitem.Item) stackitem.Item {
+	var (
+		caller = ic.VM.GetCallingScriptHash()
+		_      = toUint160(args[0])
+		amount = toBigInt(args[1])
+		pub    = toPublicKey(args[2])
+	)
+
+	if caller != n.GAS.Hash {
+		panic("only GAS is accepted")
+	}
+	if !amount.IsInt64() || amount.Int64() != n.getRegisterPriceInternal(ic.DAO) {
+		panic(fmt.Errorf("incorrect GAS amount for registration (expected %d)", n.getRegisterPriceInternal(ic.DAO)))
```

Comment:
I'd avoid two subsequent calls to `n.getRegisterPriceInternal` and use a variable instead.

Reply:
The second one is only in the case of error, not a typical scenario, but can be changed, OK.